src/main/java/studentlist/StudentList.java

```diff
@@ -14,6 +16,39 @@ public StudentList(UI ui) {
         this.ui = ui;
     }
 
+    // copy constructor
+    public StudentList(StudentList other) {
+        this.students = new ArrayList<>(other.students.size());
+        for (Student student : other.students) {
+            this.students.add(new Student(student));
+        }
+        this.ui = other.ui;
+    }
+
+    public ArrayList<Student> getList() {
+        return students;
+    }
+
+    // Method to sort students by GPA
+    public void sortStudentsByGPA() {
+        Collections.sort(students, new Comparator<Student>() {
+            @Override
+            public int compare(Student s1, Student s2) {
+                return Float.compare(s2.getGpa(), s1.getGpa());
+            }
+        });
+    }
+
+    // Method to sort students by ID
+    public void sortStudentsById() {
+        Collections.sort(students, new Comparator<Student>() {
+            @Override
+            public int compare(Student s1, Student s2) {
+                return s1.getId().compareTo(s2.getId());
+            }
+        });
+    }
+
```

Comment:
Nice use of comparator! It might be possible for this to be simplified to 1 line, but this is fine still.